File: 0004-median-of-two-sorted-arrays/solution.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int n=nums1.length;
        int m=nums2.length;
        int[] arr= new int[n+m];
        int i=0;
        for(i=0;i<n;i++){
            arr[i]=nums1[i];
        }    
        for(int j=0;j<m;j++){
            arr[i+j]=nums2[j];
        }
        Arrays.sort(arr);
        int len=m+n;
        double ans=0;
        if(len%2==0){
            int n1=len/2;
            int n2=n1-1;
            ans=(double)(arr[n1]+arr[n2])/2;
        }else{
            int n1=len/2;
            ans=(double)arr[n1];
        }
        return ans;
    }
}

File: 0004-median-of-two-sorted-arrays/solution.c
void merge(int *arr,int arrsize,int *nums,int s){
    for(int i=0;i<arrsize;i++)
    {
        arr[s]=nums[i];
        s++;
    }
}
void sort(int *arr,int n)
{
    for(int i=0;i<n;i++)
    {
        for(int j=0;j<n;j++)
        {
            if(arr[i]<arr[j])
            {
                int temp=arr[i];
                arr[i]=arr[j];
                arr[j]=temp;
            }
        }
    }
}
double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int n=nums1Size+nums2Size;
    int arr[n];
    merge(arr,nums1Size,nums1,0);
    merge(arr,nums2Size,nums2,nums1Size);
    sort(arr,n);
    printf("%d",n);
    double c=0;
    int half=n/2;
    if(n%2==0)
    {
        int a=arr[half];
        int b=arr[half-1];
        c=(double)(a+b)/2;
    }
    if(n%2!=0)
    {
        c=arr[half];
    }
    return c;
}
